### PR TITLE
Fixed swagger.yml to match API

### PR DESCRIPTION
**Pull request by dlobato**
Schema is valid now.
There's also some fixes on several object definitions to match current fields on API request/responses.

I like very much the proposed solution on https://github.com/Sylius/ShopApiPlugin/issues/251 but editing the file in one piece seems more simple.

### CODE REVIEW

File: doc/swagger.yml [hunk 1]
```diff
@@ -9,7 +9,11 @@ info:
         name: "MIT"
         url: "https://opensource.org/licenses/MIT"
 host: "demo.sylius.org"
-basePath: "/shop-api/{channelCode}"
+basePath: "/shop-api"
```
Comment (mamazu) [inline doc/swagger.yml]: Thanks for clarifying that.

Comment (dlobato) [reply doc/swagger.yml]: yep, according to the specification: 

> The basePath does not support path templating.

**Comment by dlobato** [comment]
@mamazu I did test most of the endpoints and so far I did not find any mismatch on the response codes. Anyway we are implementing a client right now, so we will very likely spot problems.

File: doc/swagger.yml (hunk 1 above)
Comment (mamazu) [reply doc/swagger.yml]: And there are some routes that don't follow that pattern I believe like the customer endpoints.

**Comment by mamazu** [comment]
Normally the response codes in the Actions would give it away but I am sure you'll find those errors as well.

**Comment by mamazu** [comment]
The current version of the documentation has different levels of indentation in different parts. But 2 spaces are the convention for yml.

**Comment by dlobato** [comment]
@lchrusciel No reason for changing. But because there was 4 and 2 on different places, I just normalized to 2.  There's still 4 spaces on some places... I'll fix it. I don't have any preference. What do you prefer?

**Comment by dlobato** [comment]
2 spaces => https://pastebin.com/dKZKPC5s
4 spaces => https://pastebin.com/ufQQX01F

**Comment by lchrusciel** [comment]
Oh, the main problem was, that it was hard to comprehend changes in the file, what has been changed. In this situation, it will not help much. Nevertheless, I would prefer 4, if you don't mind changing it.

File: doc/swagger.yml [hunk 2]
```diff
@@ -208,33 +182,26 @@ paths:
...
-          in: "path"
-          description: "Cart identifier."
-          required: true
-          type: "string"
-        - name: "identifier"
-          in: "path"
-          description: "Identifier of a specific item. Can be found in the cart summary."
-          required: true
-          type: "string"
+            -   $ref: "#/parameters/ChannelCode"
```
Comment (lchrusciel) [inline doc/swagger.yml]: Should there be so many spaces after `-` and before `$`

Comment (dlobato) [reply doc/swagger.yml]: phpstorm indent this way... I don't think there's any standard way to format yml.
For instance, https://jsonformatter.org/yaml-formatter will do:
```
...
/{channelCode}/product/{code}/reviews:
    parameters:
        -
            $ref: '#/parameters/ChannelCode'
        -
            name: code
            in: path
            description: 'Code of expected product.'
            required: true
            type: string
...
```

**Comment by lchrusciel** [comment]
Thank you, David! :1st_place_medal:

**Pull request merged**